app/assets/javascripts/issues.js

```diff
@@ -1,47 +1,158 @@
-function initIssuesMap() {
-  var map = new L.Map('issues-map', { center: new L.LatLng(48.920597, 24.709566), zoom: 14 }),
+'use strict';
+// Provide map for a map page with issues
+function initIssuesMap(hasMany, lat, lng) {
+  initGoogleLayer();
+  var map = new L.Map('map-container', { center: new L.LatLng(48.920597, 24.709566), zoom: 14 }),
       googleLayer = new L.Google('ROADMAP');
 
   map.addLayer(googleLayer);
-  
-  $.get({url: '/issues/map', dataType:"json", cache:false}).done(addMarkers);
+  // Check if page is 'issues/map'
+  if (hasMany) {  
+    $.get({url: '/issues/map', dataType:"json", cache:false}).done(function(issues){
+      addMarkers(map,issues)
+    });
+  } else {
+    // Check if page is 'Edit', 'Show' or 'New'
+    // '#issue_latitude' field specifies 'Edit' page,
+    //  where geolocation is not needed.
+    if ($("#issue_latitude").length){ 
+      editIssueCoordinates(map,lat,lng);
+    // absence of '#issue_location' field specifies 'Show' page,
+    } else if ($("#issue_location").length === 0) {
+      showIssueMap(map,lat,lng);
+    } else {
+      // Check if browser can provide geolocation
+      if (navigator.geolocation) {
+        navigator.geolocation.getCurrentPosition(function(position){
+          onGeocodeSuccess(map,position)
+        }, function(msg){
+          onGeocodeError(map,msg)
+        });
+      }
+    }
+  }
+}
```

Comment:
it was so, but codeclimate point out repeatable parts, and insists to put them all together under one function scope. How to be with it?